lib/rules/html-closing-bracket-newline.js

```diff
@@ -68,48 +75,76 @@ module.exports = {
       context.parserServices.getTemplateBodyTokenStore &&
       context.parserServices.getTemplateBodyTokenStore()
 
-    return utils.defineTemplateBodyVisitor(context, {
-      /** @param {VStartTag | VEndTag} node */
-      'VStartTag, VEndTag'(node) {
-        const closingBracketToken = template.getLastToken(node)
-        if (
-          closingBracketToken.type !== 'HTMLSelfClosingTagClose' &&
-          closingBracketToken.type !== 'HTMLTagClose'
-        ) {
-          return
-        }
+    /** @param {VStartTag | VEndTag} node */
+    function verify(node) {
+      const closingBracketToken = template.getLastToken(node)
+      if (
+        closingBracketToken.type !== 'HTMLSelfClosingTagClose' &&
+        closingBracketToken.type !== 'HTMLTagClose'
+      ) {
+        return
+      }
+
+      const prevToken = template.getTokenBefore(closingBracketToken)
+      const type =
+        node.loc.start.line === prevToken.loc.end.line
+          ? 'singleline'
+          : 'multiline'
+      const expectedLineBreaks = options[type] === 'always' ? 1 : 0
+      const actualLineBreaks =
+        closingBracketToken.loc.start.line - prevToken.loc.end.line
+
+      if (actualLineBreaks !== expectedLineBreaks) {
+        context.report({
+          node,
+          loc: {
+            start: prevToken.loc.end,
+            end: closingBracketToken.loc.start
+          },
+          message:
+            'Expected {{expected}} before closing bracket, but {{actual}} found.',
+          data: {
+            expected: getPhrase(expectedLineBreaks),
+            actual: getPhrase(actualLineBreaks)
+          },
+          fix(fixer) {
+            /** @type {Range} */
+            const range = [prevToken.range[1], closingBracketToken.range[0]]
+            const text = '\n'.repeat(expectedLineBreaks)
+            return fixer.replaceTextRange(range, text)
+          }
+        })
+      }
+    }
+
+    const documentFragment = df
 
-        const prevToken = template.getTokenBefore(closingBracketToken)
-        const type =
-          node.loc.start.line === prevToken.loc.end.line
-            ? 'singleline'
-            : 'multiline'
-        const expectedLineBreaks = options[type] === 'always' ? 1 : 0
-        const actualLineBreaks =
-          closingBracketToken.loc.start.line - prevToken.loc.end.line
+    return utils.defineTemplateBodyVisitor(
```

Comment:
I think we may be able to handle it by simply replacing it with `utils.defineDocumentVisitor`.

Reply:
Thanks, that simplifies the change a lot!